source/common/config/grpc_mux_impl.cc

```diff
@@ -217,7 +220,12 @@ void GrpcMuxImpl::onReceiveMessage(std::unique_ptr<envoy::api::v2::DiscoveryResp
           found_resources.Add()->MergeFrom(it->second);
         }
       }
-      watch->callbacks_.onConfigUpdate(found_resources, message->version_info());
+      // onConfigUpdate should be called for all watches only if message has no resources for multi
+      // watch xDS (ClusterLoadAssignment and RouteConfiguration). Otherwise only call on
```

Comment:
I think this behavior is a bit inconsistent; you don't see empty updates unless all the other EDS updates are empty, this makes per-cluster stats non-local, since they depend on other cluster EDS updates. I would prefer we just have the cluster stats `empty` be always zero for RDS/EDS.

Reply:
Yeah. Was thinking the same when I put that condition. Will change it.